promgen/views.py

```diff
@@ -1374,3 +1375,28 @@ def post(self, request, pk):
         return JsonResponse(
             {request.POST["target"]: render_to_string("promgen/ajax_clause_check.html", result)}
         )
+
+
+class PromqlQuery(View):
+    def get(self, request):
+        if not all(x in request.GET for x in ["shard", "query"]):
+            return HttpResponse("BAD REQUEST", status=400)
+
+        shard = models.Shard.objects.get(pk=request.GET["shard"])
```

Comment:
This is mostly a style thing, 
```suggestion
        shard = get_object_or_404(models.Shard, pk=request.GET["shard"])
```
but since we already have `get__object_or_404` included, this means if for whatever reason, an invalid shard ID was called, this would result in a 404 instead of a 500.

Reply:
Makes sense. Done in 04b06cc. Thanks :+1: